shortfin/setup.py

```diff
@@ -381,6 +381,27 @@ def populate_built_package(abs_dir):
         [CMakeExtension("_shortfin_default.lib")]
         + ([CMakeExtension("_shortfin_tracy.lib")] if ENABLE_TRACY else [])
     ),
+    extras_require={
+        "testing": [
+            "pytest",
+            "requests",
+        ],
+        "apps": [
+            "dataclasses-json",
+            "tokenizers",
+        ],
+        "onnx": [
+            "onnx",
+        ],
+        "torch": [
+            # Note that this pulls in `torch>=2.3.0`, which then pulls in all of CUDA.
+            "iree-turbine",
+        ],
```

Comment:
May want `onnx` to pull in `iree-compiler` and `iree-runtime` directly, then either have `torch` get them indirectly via `iree-turbine` or directly as well.

Reply:
That feels a little bit like a circular dependency as `iree-compiler[onnx]` pulls in onnx even though needed for `iree-import-onnx`:
https://github.com/iree-org/iree/blob/c86b621665ad8b898c0d0e987a45267f78eb9c3f/compiler/setup.py#L468

Reply:
Oh yeah, we could put `iree-compiler[onnx]` here instead of `onnx` on its own. I can test that.